### ISSUE TITLE
Can't run multiple Photino applications at once

### ISSUE THREAD

**Issue by cubedparadox** [tryphotino/photino.NET#188]
I've recently noticed a pretty glaring issue with Photino - if one Photino app is already running, then all subsequent Photino windows will  render a white screen.
- I assume this bug is due to the background UI web server colliding on the local port/URI it's hosting on.

Here's what happens if I launch the Photino Sample React app, followed by the Photino Sample TestBench app:
![explorer_sickcovlgz](https://github.com/tryphotino/photino.NET/assets/4484285/72929f3f-e69a-4320-b7c3-9fa673462a47)
And here's what happens if I run then in the reverse order.
![explorer_FyBXNShDgl](https://github.com/tryphotino/photino.NET/assets/4484285/b4cf9bde-33c7-433c-aa31-b71cfb654943)

I've tried a bunch of misc configuration stuff to try and work around this issue, but haven't found a good fix yet.

This occurs with the official Photino samples, my personal Photino-based project, and the [VRChat Creator Companion](https://github.com/vrchat-community/creator-companion), which is also built with Photino.

I'm filing this issue in the .NET project because that's where I've encountered it, but it's entirely possible it's a deeper Photino Native issue.

I'm testing on Windows 10, I'm happy to share additional system details and logs as needed.

**Comment by cubedparadox** [comment]
Additional Information: 
I am able to run two instances of the *same* Photino app, just not two *different* Photino apps.
![image](https://github.com/tryphotino/photino.NET/assets/4484285/245597be-156d-4ec7-b77f-01e6b73ca368)

**Comment by mksondej** [comment]
(note: not a dev of this project)

Could not reproduce this in my simple tests (built same demo app twice but with different content and it worked fine) but if you feel that the port is the issue (which is possible) then this is the part that fails here on your environment:

https://github.com/tryphotino/photino.NET.Server/blob/0a545a0c07aaa1c1fb258c91163605fdf3bfc93d/Photino.NET.Server/PhotinoServer.NET.cs#L57

As a workaround and additional test you could swap the invocation of "CreateStaticFileServer" in your project to the variant that accepts the port parameter.
In other words change this in your Program.cs:

```cs
        PhotinoServer
            .CreateStaticFileServer(args, out string baseUrl)
            .RunAsync();
```

to this:

```cs
        PhotinoServer
            .CreateStaticFileServer(args, 9000, 100, "wwwroot", out baseUrl)
            .RunAsync();
```

If that works your guess was right and we can dig further why that "first available port checking" logic fails on your end.

PS: Looking at that `if (port > port + portRange)` and that entire condition in the while loop someone must've had been in a hurry :)

**Comment by philippjbauer** [comment]
@mksondej Thank you for the reply on this issue. @cubedparadox did this resolve your issue? We couldn't reproduce the issue either.

**Comment by philippjbauer** [comment]
Closing due to inactivity.
